src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -0,0 +1,400 @@
+    KVAxesPosition axes{0u, kv_dim};
+    // (7) Make prefill model with static shapes
+    reshape_to_static(prefill_model, m_kvcache_desc.max_prompt_size, m_kvcache_desc.max_prompt_size, axes);
+    // (8) Make kvcache model with static shapes
+    reshape_to_static(kvcache_model, 1u, m_kvcache_desc.total_size, axes);
+
+    auto npudesc = extract_npu_descriptor(plugin);
+
+    auto prefill_config = pop_or_default(
+        properties_copy, "PREFILL_CONFIG", get_default_prefill_config(model, npudesc)
+    // NB: GENERATE_HINT is only applicable for default generate config!
+    );
+    auto generate_hint = str_to_hint(pop_or_default<std::string>(properties_copy, "GENERATE_HINT", "FAST_COMPILE"));
+    auto generate_config = pop_or_default(
+        properties_copy, "GENERATE_CONFIG", get_default_generate_config(model, npudesc, generate_hint)
+    );
+    merge_config_with(prefill_config, properties_copy);
+    merge_config_with(generate_config, properties_copy);
+    // FIXME: Drop CACHE_DIR option if NPUW is enabled
+    drop_cache_dir(prefill_config);
+    drop_cache_dir(generate_config);
+
+    kvcache_compiled = std::make_shared<ov::npuw::CompiledModel>(kvcache_model, plugin, generate_config);
+    prefill_compiled = std::make_shared<ov::npuw::CompiledModel>(prefill_model, plugin, prefill_config);
+
+    implement_properties();
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMCompiledModel::export_model(std::ostream& model) const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::shared_ptr<const ov::Model> ov::npuw::LLMCompiledModel::get_runtime_model() const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+void ov::npuw::LLMCompiledModel::set_property(const ov::AnyMap& properties) {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+ov::Any ov::npuw::LLMCompiledModel::get_property(const std::string& name) const {
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    auto&& configIterator = m_prop_to_opt.find(name);
+    if (configIterator != m_prop_to_opt.cend()) {
+        return std::get<1>(configIterator->second)(m_cfg);
+    } else {
+        return prefill_compiled->get_property(name);
+    }
+    OPENVINO_SUPPRESS_DEPRECATED_END
+}
+
+std::shared_ptr<ov::ISyncInferRequest> ov::npuw::LLMCompiledModel::create_sync_infer_request() const {
+    auto* non_const_this = const_cast<ov::npuw::LLMCompiledModel*>(this);  // because of const in API
+    return non_const_this->create_llm_infer_request();
+}
+
+std::shared_ptr<ov::ISyncInferRequest> ov::npuw::LLMCompiledModel::create_llm_infer_request() {
+    auto this_sptr = std::static_pointer_cast<ov::npuw::LLMCompiledModel>(shared_from_this());
+    return std::make_shared<ov::npuw::LLMInferRequest>(this_sptr, m_kvcache_desc);
+}
+
+std::shared_ptr<ov::IAsyncInferRequest> ov::npuw::LLMCompiledModel::create_infer_request() const {
+    auto internal_request = create_sync_infer_request();
+    return std::make_shared<ov::IAsyncInferRequest>(internal_request, get_task_executor(), get_callback_executor());
+}
```

Comment:
Merge it into one method please, there's really no sense in having two.

Not sure which particular overload you need, but most likely having only `create_sync_infer_request()` is enough

Reply:
Fixed

Reply:
Returned back, `create_sync_infer_request` is protected but pure virtual and `create_infer_reques` is what we want user to use

Reply:
You only need to supply `create_sync_infer_request`, the rest is managed for you.